Bug: react required as Dependency for @lexical/extension

Since the Introduction of DecoratorTextExtension react is needed to compile.

Lexical version: 0.41.0 and above

## Steps To Reproduce
1. Project without react
2. with lexical and lexical/extension
3. (An extension is used somewhere in the project)

## The current behavior
Compile fails with the error 
node_modules/@lexical/extension/DecoratorTextExtension.d.ts:9:26 - error TS2307: Cannot find module 'react' or its corresponding type declarations.
import type { JSX } from 'react';

React is not listed as peer-Dependency for lexical/extension

## The expected behavior
@lexical/extension does not require react to compile

## Impact of fix
A fix would enable projects without react that use lexical/extensions to upgrade to versions of lexical above 0.40.0


If there is some other Workaround please let me know. We do not want to require react in our application.

--- a/packages/lexical-extension/flow/LexicalExtension.js.flow
+++ b/packages/lexical-extension/flow/LexicalExtension.js.flow
@@ -191,7 +191,7 @@ export type SerializedDecoratorTextNode = {
   format: number,
   ...
 };
-declare export class DecoratorTextNode<T> extends DecoratorNode<T> {
+declare export class DecoratorTextNode extends DecoratorNode<mixed> {
   createDOM(): HTMLElement;
   getFormat(): number;
   getFormatFlags(type: TextFormatType, alignWithFormat: null | number): number;
@@ -203,7 +203,7 @@ declare export class DecoratorTextNode<T> extends DecoratorNode<T> {
 
 declare export function $isDecoratorTextNode(
   node: ?LexicalNode,
-): node is DecoratorTextNode<{...}>;
+): node is DecoratorTextNode;
 
 declare export var DecoratorTextExtension: LexicalExtension<ExtensionConfigBase, "@lexical/extension/DecoratorText", void, void>;
 

--- a/packages/lexical-extension/src/DecoratorTextExtension.ts
+++ b/packages/lexical-extension/src/DecoratorTextExtension.ts
@@ -14,7 +14,6 @@ import type {
   StateValueOrUpdater,
   TextFormatType,
 } from 'lexical';
-import type {JSX} from 'react';
 
 import {
   $getSelection,
@@ -42,7 +41,7 @@ const formatState = createState('format', {
   parse: (value) => (typeof value === 'number' ? value : 0),
 });
 
-export class DecoratorTextNode extends DecoratorNode<JSX.Element> {
+export class DecoratorTextNode extends DecoratorNode<unknown> {
   $config() {
     return this.config('decorator-text', {
       extends: DecoratorNode,

